language-server/modules/langserver-core/src/test/resources/completion/expression_context/source/list_constructor_ctx_source6.bal

```diff
@@ -0,0 +1,32 @@
+int[] arr = [1, 2, 3];
+
+int[][] dim2Arr = [[1, 2, 3]];
+
+int i = 10;
+
+string a = "Hello";
+
+type Dim1 int[];
+
+public function main() {
+
+    int[] intArr1 = [];
+
+    int[] intArr2 = [...arr, ];
+
+    int[][] intArr3 = [[]];
+
+    int[][] intArr4 = [...dim2Arr, ];
+
+    Dim1[] intArr5 = [];
+
+    int[] intArr6 = [...];
+}
+
+function getArray() returns int[] {
+
+}
+
+function getInt() returns int {
+
+}
```

Comment:
```ballerina
function foo() returns int[]|error {
    return [1, 2];
}

public function main() returns error? {
    int[] arr = [<cursor>];
}
```
For this case we do not suggest `check` with `foo()`. Can we do that?

Reply:
You mean adding a completion item like `check foo()`

Reply:
Consider the following case 
![image](https://github.com/ballerina-platform/ballerina-lang/assets/35211477/5f93bb8f-6cf2-4c37-905d-8e8d8cf28003)

We are not suggesting `check` with the expression in the above case as well. However, we suggest the `add check` code action to check the error.

![image](https://github.com/ballerina-platform/ballerina-lang/assets/35211477/47f1de71-5aa0-4b2b-8cba-59a720668359)

In the case above we don't get the code action. Created https://github.com/ballerina-platform/ballerina-lang/issues/41165 to track this